File: CPP/program/Garage.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Parking.cpp"
#include "Car.cpp"
#include "Motorcycle.cpp"


// Deklarasi namespace
using namespace std;


class Garage : public Parking{
private:
    string tipePintu;
    string pemilik;
    
public:
    // Constructor
    Garage()
    {
    }
    
    // Constructor dengan parameter lengkap
    Garage(string tipePintu, string pemilik, string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Car> mobil, vector<Motorcycle> motor) : Parking(nama, kapasitas, luas, jumlahKendaraanSaatIni, mobil, motor){
        this->tipePintu = tipePintu;
        this->pemilik = pemilik;
    }
    // Constructor dengan parameter lengkap
    Garage(string tipePintu, string pemilik, string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Car> mobil) : Parking(nama, kapasitas, luas, jumlahKendaraanSaatIni, mobil){
        this->tipePintu = tipePintu;
        this->pemilik = pemilik;
    }
    // Constructor dengan parameter lengkap
    Garage(string tipePintu, string pemilik, string nama, int kapasitas, int luas, int jumlahKendaraanSaatIni, vector<Motorcycle> motor) : Parking(nama, kapasitas, luas, jumlahKendaraanSaatIni, motor){
        this->tipePintu = tipePintu;
        this->pemilik = pemilik;
    }


    // Getter untuk mengambil nilai atribut tipePintu
    string getTipePintu() {
        return tipePintu;
    }

    // Setter untuk mengatur nilai atribut tipePintu
    void setTipePintu(string tipePintu) {
        this->tipePintu = tipePintu;
    }

    // Getter untuk mengambil nilai atribut pemilik
    string getPemilik() {
        return pemilik;
    }

    // Setter untuk mengatur nilai atribut pemilik
    void setPemilik(string pemilik) {
        this->pemilik = pemilik;
    }

    
    // Destructor
    ~Garage()
    {

    }
};

File: Python/program/Garage.py
# Kelas Garasi
from Parking import Parking

class Garage(Parking):
    __tipePintu=""
    __pemilik=""
    
    # Constructor dengan parameter lengkap
    def __init__(self,  tipePintu="", pemilik="", nama="", kapasitas=0, luas=0, jumlahKendaraanSaatIni=0, listMobil=[], listMotor=[]) : 
        super().__init__(nama, kapasitas, luas, jumlahKendaraanSaatIni, listMobil, listMotor)
        self.__tipePintu = tipePintu
        self.__pemilik = pemilik

    # Getter untuk mengambil nilai atribut tipePintu
    def getTipePintu(self) :
        return self.__tipePintu

    # Setter untuk mengatur nilai atribut tipePintu
    def setTipePintu(self, tipePintu) :
        self.__tipePintu = tipePintu

    # Getter untuk mengambil nilai atribut pemilik
    def getPemilik(self) :
        return self.__pemilik

    # Setter untuk mengatur nilai atribut pemilik
    def setPemilik(self, pemilik) :
        self.__pemilik = pemilik